### DISCUSSION TITLE
How to rewrite properly YouTube urls

### DISCUSSION THREAD

**Discussion by azlekov** [FreeTubeApp/FreeTube#2747] [Q&A]
Hello,

I'm using v0.17.1 Beta on macOS

I did check:
- https://github.com/SimonBrazell/privacy-redirect/blob/b31c5bf75a8e7dfef9ef5e55b8479bd553978578/src/pages/background/background.js#L277
- https://github.com/libredirect/libredirect/blob/4709a6667d71c0e57c0de8664d106be073a338db/src/assets/javascripts/services.js#L164

But when I paste rewritten url in safari for example, it prompts to open in FreeTube, but the video does not start. I see just the default home page.

```
freetube://https://youtu.be/y9TizOUqsuc
```

What I'm doing wrong? I cannot find documentation how to rewrite urls and what schemas are supported by FreeTube

Thanks!

**Comment by absidue** [comment]
Thanks for reporting the bug, we were able to replicate it.
This seems to be a macOS specific bug that doesn't exist in the Linux or Windows builds of FreeTube (macOS implements URL schemes differently than Windows and Linux).
We'll look into fixing it.

**Reply by azlekov** [reply]
Is this a know issue @absidue?

If this is helping I tried Browserosaurus and the rewrite there or how they did it works: 
https://github.com/will-stone/browserosaurus/blob/8e0e315216aec097388491c8695dba806d48f61f/src/config/apps.ts#L90

**Reply by azlekov** [reply]
Looks like Browserosaurus are open the app with a parameter to the CLI: https://github.com/will-stone/browserosaurus/blob/5418811dcd6abf214f01625ff88f04f4f9546107/src/main/utils/open-app.ts#L31

**Reply by absidue** [reply]
Now that you've reported it, it is a known issue.

Yes, they are taking advantage of the code in FreeTube that handles the way that URL schemes work in Windows and Linux. Regardless of what 3rd party tools do, we do still need to fix the handling of the macOS native way of doing it.

If you are interested in taking a look at it yourself, the relevant lines in FreeTube's code are here:
[src/main/index.js#L738-L748](https://github.com/FreeTubeApp/FreeTube/blob/development/src/main/index.js#L738-L748)

[src/renderer/App.js#L501-L505](https://github.com/FreeTubeApp/FreeTube/blob/development/src/renderer/App.js#L501-L505)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@PikachuEXE maybe something for u to tackle?

**Reply by PikachuEXE** [reply]
What are the steps to reproduce?
- Install [libredirect](https://github.com/libredirect/libredirect) ([privacy-redirect](https://github.com/SimonBrazell/privacy-redirect) doesn't seem maintained anymore)
- Open a video in browser
- Profit?

Does it have to be Safari or any Chromium based browser also has the same issue? (Let me test this)

**Reply by PikachuEXE** [reply]
Using MacOS 12.6.1, Vivaldi 5.5.2805.42, libredirect 2.3.4, FreeTube nightly build (I can test on 0.18.0 if you can reproduce the issue)
Opening a video like https://youtu.be/y9TizOUqsuc in browser does redirect and FT would navigate to the video view for that video properly

**Reply by efb4f5ff-1298-471a-8973-3d47447115dc** [reply]
> What are the steps to reproduce?
> - Install [libredirect](https://github.com/libredirect/libredirect) ([privacy-redirect](https://github.com/SimonBrazell/privacy-redirect) doesn't seem maintained anymore)
> - Open a video in browser
> - Profit?
> 
> Does it have to be Safari or any Chromium based browser also has the same issue? (Let me test this)

@absidue @azlekov maybe u can help this poor pika out

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Opened https://github.com/FreeTubeApp/FreeTube/issues/3257 to track this.
